test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/KafkaE2EFixture.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common
+{
+	/* Common fixture for all language test case classes which does -
+	* Azure Infra setup and Func Apps Startup
+	* Stopping Func Apps and Azure Infra cleanup
+	*/
+	public class KafkaE2EFixture : IAsyncLifetime
+	{
+		private Language _language;
+		private AppType _appType;
+		private BrokerType _brokerType;
+		protected bool isInitialized = false;
+		private readonly ILogger _logger = TestLogger.GetTestLogger();
+
+		public KafkaE2EFixture()
+		{
+			_logger.LogInformation($"Kafkae2efixture for {_language} {_appType} {_brokerType}");
+		}
+
+		public void OrchestrateInitialization()
+		{
+			if (isInitialized)
+			{
+				return;
+			}
+
+			//Azure Infra setup and Func Apps Startup
+			TestSuitInitializer testSuitInitializer = new();
+			testSuitInitializer.InitializeTestSuit(_language, _brokerType);
+
+			isInitialized = true;
+		}
+
+		async Task IAsyncLifetime.DisposeAsync()
+		{
+			//Stopping Func Apps and Azure Infra cleanup
+			TestSuiteCleaner testSuitCleaner = new TestSuiteCleaner();
```

Comment:
Can we have this as single class like for eg. TestSuiteInfratructureBuilder or TestInfrastructureManager and then have Initialize(or create or build) and Destroy( or cleanup ,etc) so that the class which creates it is responsible to destroy it.

Reply:
In xunit, this is the responsibility of the fixture class. This KafkaE2EFixture class does exactly what you are asking for. The only difference being that it gives the responsibility of initialization to a separate class and cleanup to a separate class. This allows for more flexibility in the sense that the implementation for initialization and cleanup need not be tied with one another.

Reply:
Let me call you and explain what i meant here.